File: crates/deslop/tests/fixtures/incremental-multilang/src/ledger_alpha.ts
// ledger_alpha.ts — the canonical copy of the TypeScript reconciliation routine.

export const ALPHA_LEDGER_TAG = "alpha";

export function reconcileEntries(entries: number[], floor: number): number {
  let balance = 0;
  for (const entry of entries) {
    if (entry > floor) {
      balance += entry * 2;
    } else {
      balance -= Math.trunc(entry / 2);
    }
  }
  return balance;
}

File: crates/deslop/tests/fixtures/incremental-multilang/src/ledger_alpha.dart
// ledger_alpha.dart — the canonical copy of the Dart reconciliation routine.

const String alphaLedgerTag = 'alpha';

int reconcileEntries(List<int> entries, int floor) {
  var balance = 0;
  for (final entry in entries) {
    if (entry > floor) {
      balance += entry * 2;
    } else {
      balance -= entry ~/ 2;
    }
  }
  return balance;
}
